girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -106,6 +107,7 @@ def __init__(self, assetstore):
         # happens to existing assetstores that no longer can access their temp
         # directories.
         self.tempDir = os.path.join(self.assetstore['root'], 'temp')
+        self.deleteLock = threading.Lock()
```

Comment:
Hm, this should still help shorten the critical section, but we can remove all the lock stuff.

In general, we can't use this locking primitive to assume we are locked across threads, since we have to assume there could be many app servers.